File: algorithms/implementation/the-hurdle-race/the-hurdle-race-python/tests/test_hurdleRace.py
from hurdleRace import hurdleRace

class TesteHurdleRace:
    
    def testeCaso1(self):
        
        k = 4
        altura = [1, 6, 3, 5, 2]
        
        esperado = 2
        
        resultado = hurdleRace(k, altura)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        k = 7
        altura = [2, 5, 4, 5, 2]
        
        esperado = 0
        
        resultado = hurdleRace(k, altura)
        
        assert resultado == esperado

File: algorithms/implementation/the-hurdle-race/the-hurdle-race-javascript/js/test_hurdleRace.js
import {hurdleRace} from '../js/hurdleRace.js';
import assert from 'assert';

describe('hurdleRace', function() {

    it('O número mínimo de doses necessárias para o salto é 2', function() {

        let k = 4;
        let altura = [1, 6, 3, 5, 2];

        let resultado = hurdleRace(k, altura);

        let esperado = 2;

        assert.strictEqual(resultado, esperado)
    });
});

describe('hurdleRace', function() {

    it('O número mínimo de doses necessárias para o salto é 0', function() {

        let k = 7;
        let altura = [2, 5, 4, 5, 2];

        let resultado = hurdleRace(k, altura);

        let esperado = 0;

        assert.strictEqual(resultado, esperado)
    });
});